### PR TITLE
add fog clock globe components

### PR DESCRIPTION
**Pull request by kurohune538**
- [x] Fog
- [x] Clock
- [x] Globe

### CODE REVIEW

File: src/Globe.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  nightFadeInDistance?: number;
+  nightFadeOutDistance?: number;
+  oceanNormalMapUrl?: string;
+  shadows?: Cesium.ShadowMode;
+  show?: boolean;
+  showGroundAtmosphere?: boolean;
+  showWaterEffect?: boolean;
+  terrainProvider?: Cesium.TerrainProvider;
+  tileCacheSize?: number;
+  tileLoadProgressEvent?: Cesium.Event;
```
Comment (rot1024) [inline src/Globe.ts]: this is also a cesium event.

File: src/Clock.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import Cesium, { CesiumWidget } from "cesium";
+
+export interface ClockCesiumProps {
+  canAnimate?: boolean;
+  clockRange?: Cesium.ClockRange;
+  clockStep?: Cesium.ClockStep;
+  currentTime?: Cesium.JulianDate;
+  multiplier?: number;
+  onStop?: Cesium.Event;
+  onTick?: Cesium.Event;
```
Comment (rot1024) [inline src/Clock.ts]: Please separate them into cesiumEventProps

File: src/Globe.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  tileLoadedEvent: "onTileLoad",
+};
+
+export interface GlobeProps
+  extends GlobeCesiumProps,
+    GlobeCesiumReadonlyProps,
+    GlobeCesiumEventProps {}
+
+export interface GlobeContext {
+  scene: Cesium.Scene;
```
Comment (rot1024) [inline src/Globe.ts]: make this optional

```
scene?: Cesium.Scene
```

File: src/Clock.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  shouldAnimate?: boolean;
+  startTime?: Cesium.JulianDate;
+  stopTime?: Cesium.JulianDate;
+}
+
+/* tslint:disable-next-line: no-empty-interface */
+export interface ClockProps extends ClockCesiumProps {}
+
+/* tslint:disable-next-line: no-empty-interface */
```
Comment (rot1024) [inline src/Clock.ts]: unnecessary comment

File: src/Fog.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  density?: number;
+  enabled?: boolean;
+  minimumBrightness?: number;
+  screenSpaceErrorFactor?: number;
+}
+
+/* tslint:disable-next-line: no-empty-interface */
+export interface FogProps extends FogCesiumProps {}
+
+/* tslint:disable-next-line: no-empty-interface */
```
Comment (rot1024) [inline src/Fog.ts]: unnecessary comment

File: src/Globe.ts (hunk 1 above)
Comment (kurohune538) [reply src/Globe.ts]: done!

File: src/Clock.ts (hunk 2 above)
Comment (kurohune538) [reply src/Clock.ts]: done!

File: src/Clock.ts (hunk 4 above)
Comment (kurohune538) [reply src/Clock.ts]: OK I'll remove them

File: src/Fog.ts (hunk 5 above)
Comment (kurohune538) [reply src/Fog.ts]: same as upon

File: src/Globe.ts [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  shadows?: Cesium.ShadowMode;
+  show?: boolean;
+  showGroundAtmosphere?: boolean;
+  showWaterEffect?: boolean;
+  terrainProvider?: Cesium.TerrainProvider;
+  tileCacheSize?: number;
+}
+
+export interface GlobeCesiumReadonlyProps {
+  tilesLoaded?: boolean;
```
Comment (rot1024) [inline src/Globe.ts]: unnecessary prop

**Pull request merged**